packages/evm/test/precompiles/eip-2537-bls.spec.ts

```diff
@@ -35,48 +36,63 @@ const precompileMap: { [key: string]: string } = {
 
 const common = new Common({ chain: Chain.Mainnet, hardfork: Hardfork.Berlin, eips: [2537] })
 
-for (const fname of files) {
-  const fullName = `${dir}/${fname}`
-  const parsedJSON = JSON.parse(readFileSync(fullName, 'utf-8'))
-  describe(`Precompiles: ${fname}`, () => {
-    for (const data of parsedJSON) {
-      it(`${data.Name}`, async () => {
-        const evm = await EVM.create({
-          common,
-        })
-        const precompileAddr = precompileMap[fname]
-        const precompile = getActivePrecompiles(common).get(precompileAddr)!
-
-        const callData = {
-          data: hexToBytes(`0x${data.Input}`),
-          gasLimit: BigInt(5000000),
-          common,
-          _EVM: evm,
-        }
-
-        if (data.ExpectedError !== undefined) {
-          try {
-            await precompile(callData)
-            assert.fail('The precompile should have thrown')
-          } catch {
-            // If the precompile throws, catch it here, but this is fine, since it is expected to fail
+// MCL Instantiation
+await mcl.init(mcl.BLS12_381)
+mcl.setMapToMode(mcl.IRTF) // set the right map mode; otherwise mapToG2 will return wrong values.
+mcl.verifyOrderG1(true) // subgroup checks for G1
+mcl.verifyOrderG2(true) // subgroup checks for G2
+const mclbls = new MCLBLS(mcl)
```

Comment:
I think we can maybe improve here (I also now see why we export MCLBLS / NobleBLS). Can we add a static method to the interfaces which inits these BLSs? Then we can init the MCL using `MCLBLS.init(mcl)` and write those settings there

Reply:
That's a great idea, will directly add! 👍 🙂

Reply:
I now went for a non-static init() version, I find it a bit nicer if the out asny initialization (so `await mcl.init(mcl.BLS12_381)`) is not also wrapped and can be done separately by the user, guess both ways would work and be fine though, for this one I find the typing also a bit nicer. The optional `init()` method from the interface is the automatically called from within the EVM constructor if present.

```ts
import * as mcl from 'mcl-wasm'

await mcl.init(mcl.BLS12_381)
const evm = await EVM.create({ mcl: new MCLBLS(mcl) })
```